packages/browser/src/plugins/in-app-plugin/__tests__/index.test.ts

```diff
@@ -0,0 +1,138 @@
+import { Analytics } from '../../../core/analytics'
+import { Plugin } from '../../../core/plugin'
+import { pageEnrichment } from '../../page-enrichment'
+import { customerio, CustomerioSettings } from '../../customerio'
+import { InAppPlugin, InAppPluginSettings } from '../'
+import cookie from 'js-cookie'
+import Gist from 'customerio-gist-web'
+
+jest.mock('unfetch', () => {
+  return jest.fn()
+})
+
+describe('Customer.io In-App Plugin', () => {
+  let options: CustomerioSettings
+  let settings: InAppPluginSettings
+  let analytics: Analytics
+  let inAppPlugin: Plugin
+  let cio: Plugin
+
+  beforeEach(async () => {
+    jest.resetAllMocks()
+    jest.restoreAllMocks()
+
+    options = { apiKey: 'foo' }
+    analytics = new Analytics({ writeKey: options.apiKey })
+    cio = customerio(analytics, options, {})
+
+    settings = {
+      siteId: 'siteid',
+    }
+    inAppPlugin = InAppPlugin(settings)
+
+    await analytics.register(inAppPlugin, pageEnrichment)
+
+    window.localStorage.clear()
+  })
+
+  function resetCookies(): void {
+    Object.keys(cookie.get()).map((key) => cookie.remove(key))
+  }
+
+  afterEach(async () => {
+    analytics.reset()
```

Comment:
I'd recommend placing this in the `beforeEach`. Tests that fail will not execute the afterEach so you might get subsequent failures due to state bleeding over.